dask/array/utils.py

```diff
@@ -108,7 +109,9 @@ def meta_from_array(x, ndim=None, dtype=None):
 
 
 def compute_meta(func, _dtype, *args, **kwargs):
-    with np.errstate(all="ignore"):
+    with np.errstate(all="ignore"), warnings.catch_warnings():
+        warnings.simplefilter("ignore", category=RuntimeWarning)
```

Comment:
I'm less sure of my understanding of warnings handling. 

@jcrist, does this look ok to you?  (this review should take only a minute or two I think)

Reply:
This looks correct, but maybe we should silence all warnings instead of just runtime warnings? Not sure.

Reply:
I have a slight preference for just ignoring `RuntimeWarning` for now and broadening to all warnings if the need comes up. But it's probably not a big deal either way.